src/components/header-footer.js

```diff
@@ -37,7 +38,48 @@ function Header() {
 }
 
 
-function Footer() {
+function UserHeader() {
```

Comment:
For the nonUserHeader and userHeader why not check for the token in and only have one header / footer component that changes based on if the token is present or not.